### PR TITLE
fixes #102 Browsing with tabulation in Carousel can switch on hidden cards with buttons

### LINKED ISSUE #102: Browsing with tabulation in Carousel can switch on hidden cards with buttons
Buttons on hidden cards of carousel can be accessed when an user tab. This is a problem for accessibility, only visible cards of the Carousel should be accessed by keyboard browsing.

image

Possible solution : Buttons on hidden cards must have tabIndex="-1"  as attribute.

### PR DESCRIPTION
**Pull request by correi-f**
fixes #102 Browsing with tabulation in Carousel can switch on hidden cards with buttons

### CODE REVIEW

File: src/components/Carousel/hooks/useCarousel.ts [hunk 1]
```diff
@@ -36,9 +74,11 @@ function setAriaAttributes(
       ) {
         item.current?.setAttribute('aria-hidden', 'true');
         item.current?.setAttribute('tabIndex', '-1');
+        setButtonsFromCardTabIndex(item.current, true);
```
Comment (Fabilin) [inline src/components/Carousel/hooks/useCarousel.ts:77]: Based on React conventions, I believe this should be implemented as state on the individual cards, updating the tabIndex in the re-render.

Comment (correi-f) [reply src/components/Carousel/hooks/useCarousel.ts]: Done

File: src/components/Card/Card.tsx [hunk 2]
```diff
@@ -154,12 +163,23 @@ const Card = React.forwardRef<HTMLLIElement, CardProps>(function cardRender(
           <ButtonList>
             {buttons.map((button, index) => (
               <li key={index}>
-                <Button
-                  onClick={onAction.bind(null, button)}
-                  onKeyPress={onAction.bind(null, button)}
-                >
-                  {button.label}
-                </Button>
+                {isHidden && (
```
Comment (elebescond) [inline src/components/Card/Card.tsx:166]: Nice to have. Possibilité de limiter le test à l'attribut tabIndex pour éviter la duplication de code

**Review by elebescond — APPROVED** [review]

File: src/components/Card/Card.tsx (hunk 2 above)
Comment (correi-f) [reply src/components/Card/Card.tsx]: Done

**Review by Fabilin — APPROVED** [review]

**Review by francoisno — APPROVED** [review]

**Pull request merged**